### PR TITLE
Gwo - ready for release

### PR DESCRIPTION
**Pull request by Laura-Jane**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { Point, ResultMessage, StartMessage } from "./message.interface";
+import { Position } from "./Point";
+import { Wolf } from "./wolf";
+import { HuntingPrey } from "./HuntingPrey";
+
+export declare function addEventListener(event: string, handler: (any) => void): void;
+export declare function postMessage(data: any);
+
+function rateWolves(wolves: Wolf[]): Wolf[] {
+    let fitnessValues = [];
```
Comment (raspyweather) [inline src/index.ts]: use const if possible (should be as far as I see)

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+function runCode(): Point[] {
+    console.log('runcode');
+
+    let wolves: Wolf[] = [];
+    let wolfAlpha: Wolf;
+    let wolfBeta: Wolf;
+    let wolfDelta: Wolf;
+    let a: number;
```
Comment (raspyweather) [inline src/index.ts]: is there a better variable name available?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            console.log([
+                lastResult, bestPosition,
+                lastResult && bestPosition && lastResult.subtract(bestPosition.x, bestPosition.y),
+                lastResult && bestPosition && !lastResult.compareToPoint(bestPosition, minimumEpsilon)]
+            );
+        }
+
+        let huntingPrey = new HuntingPrey(wolfAlpha, wolfBeta, wolfDelta, a, searchDomain);
+        
+        wolves.forEach(wolf => { huntingPrey.hunt(wolf); });
```
Comment (raspyweather) [inline src/index.ts]: curly braces not needed ☺

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        iterationCounter++;
+    }
+
+    return results;
+}
+
+addEventListener('message', (message: { data: any }) => {
+    try {
+        let data = <StartMessage>message.data;
```
Comment (raspyweather) [inline src/index.ts]: use const

File: src/wolf.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Position } from "./Point";
+
+export class Wolf {
+    private position: Position;
+    private functionToOptimize: (x: number, y: number) => number;
```
Comment (raspyweather) [inline src/wolf.ts]: use readonly for immutability

File: src/index.ts (hunk 1 above)
Comment (Laura-Jane) [reply src/index.ts]: let -> const here

File: src/index.ts (hunk 2 above)
Comment (Laura-Jane) [reply src/index.ts]: a got a new name: seekOrHuntCoefficient

File: src/index.ts (hunk 3 above)
Comment (Laura-Jane) [reply src/index.ts]: i removed them :)

File: src/index.ts (hunk 4 above)
Comment (Laura-Jane) [reply src/index.ts]: used const now (I just copied this from cso, so I will change it there, too)

File: src/wolf.ts (hunk 5 above)
Comment (Laura-Jane) [reply src/wolf.ts]: readonly added to functionToOptimize

File: src/HuntingPrey.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ export class HuntingPrey implements IHuntingPrey {
     constructor(private readonly wolfAlpha: Wolf, 
                 private readonly wolfBeta: Wolf, 
                 private readonly wolfDelta: Wolf, 
-                private readonly a: number, 
+                private readonly seekOrHuntCoefficient: number,
```
Comment (raspyweather) [inline src/HuntingPrey.ts]: I approve this 👍

**Pull request merged**